networkx/algorithms/tree/tests/test_mst.py

```diff
@@ -261,6 +292,26 @@ class TestPrim(MultigraphMSTTestBase):
 
     algorithm = "prim"
 
+    def test_ignore_nan(self):
+        """Tests that the edges with NaN weights are ignored or
+        raise an Error based on ignore_nan is true or false"""
+        H = nx.MultiGraph()
```

Comment:
```suggestion
        pytest.importorskip("numpy")
        H = nx.MultiGraph()
```

Reply:
When I do this, tests are failing on my local system because NaN is not recognized. I have tried two ways -

![image](https://user-images.githubusercontent.com/82928853/224942051-d5d7d92a-ca10-4ce7-8ad5-40449077268b.png)

![image](https://user-images.githubusercontent.com/82928853/224942094-77f07c77-55ef-4d1c-8bdc-cf6e111a7f49.png)

Reply:
@MridulS I realized that I actually didn't have to use Numpy for NaN edge weight, I used `weight=float("nan")` instead and the test cases passed alright on my local system. However, when I push the changes, the checks show that ValueError is never raised. But when I run the example, ValueError is getting raised.

<img width="819" alt="image" src="https://user-images.githubusercontent.com/82928853/224955552-df2c3dd1-0aec-48ed-b8dd-e055a8d3abce.png">

I cannot figure out what the problem is here.

Reply:
The difference is that the test is being run using Pypy instead of CPython. The test suite has many different configurations and running it on this alternative version of python is one of them. Perhaps they handle `math.isnan` differently. But I haven't looked.

Also, could you cut/paste text into github rather than images where possible -- so that we can cut/paste from github into our python sessions? To get the code to look nice use triple-backticks like ` ``` ` and when it is python code, you can get it highlighted nicely using a paragraph like this:
```
```python
H = nx.Graph([(i, i+1) if i>3 else (i+100, 0)])
``
```
Click on the Preview tab (of the comment) to check the formatting, and then on the Write tab to edit it.

Thanks for these PRs.

Reply:
@dschult I have added the code for the last comment, and will keep this format in mind for the future as well!

Reply:
Well...   I installed pypy using conda and it seems to raise the ValueError as we want it to.
Not sure why the test is failing.

Reply:
Hah Haa!  Adding this test caught an error. And it's because we use pypy in our tests.
Apparently, pypy set.pop() produces the items in a different order from set.pop() in Cpython. That mixed up the order that the nodes are processed in `minimum_spanning_edges` in the prim algorithm. That bug that this caught was that edges from nodes `u` (found as roots) are checked for `nan` near line 296. But edges from the neighbors v of those nodes are explored later in the code: near line 336 and 342.

Can you add the code that catches `nan` just after each of those `d.get("weight", 1)` lines?

Hopefully that will fix the trouble.